auth-source-pass.el

```diff
@@ -265,33 +235,105 @@ inside USER by giving priority to USER.  Same for PORT."
      (or port (number-to-string (url-port url))))))
 
 (defun auth-source-pass--find-match-unambiguous (hostname user port)
-  "Return a password-store entry name matching HOSTNAME, USER and PORT.
+  "Return password-store entry data matching HOSTNAME, USER and PORT.
 If many matches are found, return the first one.  If no match is found,
 return nil.
 
 HOSTNAME should not contain any username or port number."
-  (or
-   (and user port (auth-source-pass--find-one-by-entry-name
-                   (format "%s@%s%s%s" user hostname auth-source-pass-port-separator port)
-                   user))
-   (and user port (auth-source-pass--find-one-by-entry-name
-                   (format "%s%s%s" hostname auth-source-pass-port-separator port)
-                   user))
-   (and user (auth-source-pass--find-one-by-entry-name
-              (format "%s@%s" user hostname)
-              user))
-   (and port (auth-source-pass--find-one-by-entry-name
-              (format "%s%s%s" hostname auth-source-pass-port-separator port)
-              nil))
-   (auth-source-pass--find-one-by-entry-name hostname user)
-   ;; if that didn't work, remove subdomain: foo.bar.com -> bar.com
-   (let ((components (split-string hostname "\\.")))
-     (when (= (length components) 3)
-       ;; start from scratch
-       (auth-source-pass--find-match-unambiguous
-        (mapconcat 'identity (cdr components) ".")
-        user
-        port)))))
+  (cl-reduce
+   (lambda (result entry-set)
+     (or result
+         (pcase (length entry-set)
+           (0 (auth-source-pass--do-debug "no match found")
```

Comment:
What information would you like to capture in the log, the possibilities that were actually found in the store, or all of the possibilities that were evaluated for whether they were in the store? What do you think about `--applicable-entries` just logging the flattened version of the result it would return or "no match found" if that is `nil`?

Note that changing this logging will also require changes to some of the tests.

Reply:
What I would like is the list of possible suffixes, then the list of matching entries then the chosen one.